pkg/didcomm/protocol/decorator/decorator.go

```diff
@@ -118,7 +127,116 @@ func (d *AttachmentData) Fetch() ([]byte, error) {
 
 	// TODO add support to fetch links
 
-	// TODO add support for jws signatures
-
 	return nil, errors.New("no contents in this attachment")
 }
+
+type rawSig struct {
+	Header    rawSigHeader `json:"header,omitempty"`
+	Signature string       `json:"signature,omitempty"`
+	Protected string       `json:"protected,omitempty"`
+}
+
+type rawSigHeader struct {
+	KID string `json:"kid,omitempty"`
+}
+
+type rawProtected struct {
+	JWK json.RawMessage `json:"jwk,omitempty"`
+	Alg string          `json:"alg,omitempty"`
+}
+
+// Sign signs the base64 payload of the AttachmentData, and adds the signature to the attachment.
+func (d *AttachmentData) Sign(c crypto.Crypto, kh interface{}, pub []byte) error {
+	didKey, _ := fingerprint.CreateDIDKey(pub)
+
+	jwk, err := jose.JWKFromKey(ed25519.PublicKey(pub))
```

Comment:
Please don't hard code key types.

cc @Baha-sk

Reply:
Right now, aca-py hardcodes to ed25519, but you're right, key type choice is made by the caller and `Sign()` should respect it.

Reply:
if this is only for interop, then it's not a big deal as long as it will be removed when V1 interop is not needed, otherwise, this function has to be able to support signing with P-256 and P-384 keys as well

Reply:
`AttachmentData` is a standard DIDComm message structure not exclusive to interop.

Reply:
Agree with @llorllale - please don't hardcode key type.

Reply:
This is done